src/components/select/Select.tsx

```diff
@@ -46,26 +124,43 @@ selectOverlayNode.setAttribute('data-teamleader-ui', 'select-overlay');
 
 const activeSelects = new Set();
 
-class Select extends PureComponent {
-  componentDidMount() {
+function Select<Option extends OptionType, IsMulti extends boolean, IsClearable extends boolean>(
+  {
+    components,
+    creatable = false,
+    error,
+    inverse = false,
+    helpText,
+    menuHorizontalOffset,
+    size = 'medium',
+    success,
+    warning,
+    menuWidth,
+    width = '100%',
+    truncateOptionText,
+    options,
+    ...otherProps
+  }: SelectProps<Option, IsMulti, IsClearable>,
+  ref: ForwardedRef<SelectType<Option, IsMulti>>,
+) {
+  useEffect(() => {
+    // @ts-ignore implicit type of any, nothing to do here about it
```

Comment:
You can declare what `this` is like this: https://www.typescriptlang.org/docs/handbook/2/functions.html#declaring-this-in-a-function

First argument of your `Select` function can be `this: GenericComponent<SelectProps>`